packages/tools/devtools/devtools-browser-extension/src/popup/PopupView.tsx

```diff
@@ -0,0 +1,124 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React from "react";
+import {
+	DevtoolsFeatures,
+	GetDevtoolsFeatures,
+	type ISourcedDevtoolsMessage,
+	type InboundHandlers,
+	handleIncomingMessage,
+} from "@fluidframework/devtools-core";
+import { type BackgroundConnection } from "../BackgroundConnection";
+
+const queryTimeoutInMilliseconds = 2000; // 2 seconds
+
+const loggingContext = "FLUID_DEVTOOLS(Popup)";
+
+/**
+ * Message sent to the webpage to query for the supported set of Devtools features.
+ */
+const getSupportedFeaturesMessage = GetDevtoolsFeatures.createMessage();
+
+/**
+ * Props containing critical BG service connection used for message passing with webpage
+ */
+export interface PopupViewProps {
+	backgroundServiceConnection: BackgroundConnection;
+}
+
+/**
+ * Component that renders when the user clicks the extension icon *in the browser toolbar*
+ * @returns popup component
+ */
+export function PopupView(props: PopupViewProps): React.ReactElement {
+	const { backgroundServiceConnection } = props;
+
+	// Indicates if Fluid Devtools is running in the current page (i.e. the application is using it and initialized it correctly).
+	// Undefined means still looking or that the message which is sent to discover this has not been sent yet.
+	const [foundDevtools, setFoundDevtools] = React.useState<boolean | undefined>(undefined);
+
+	React.useEffect(() => {
+		/**
+		 * Handlers for inbound messages.
+		 */
+		const inboundMessageHandlers: InboundHandlers = {
+			[DevtoolsFeatures.MessageType]: async (untypedMessage) => {
+				// We don't actually care what the supported features are here, we're just using the response
+				// as verification that Fluid Devtools are running on the page.
+				setFoundDevtools(true);
+				clearTimeout(responseTimeout);
+				return true;
+			},
+		};
+
+		/**
+		 * Event handler for messages coming from the Background Script.
+		 */
+		function messageHandler(message: Partial<ISourcedDevtoolsMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers, {
+				context: loggingContext,
+			});
+		}
+		backgroundServiceConnection.on("message", messageHandler);
+
+		// Query for supported feature set
+		backgroundServiceConnection.postMessage(getSupportedFeaturesMessage);
+		let responseTimeout: NodeJS.Timeout | undefined;
+		if (foundDevtools === undefined) {
+			responseTimeout = setTimeout(() => {
+				setFoundDevtools(false);
+			}, queryTimeoutInMilliseconds);
+		}
+
+		return (): void => {
+			backgroundServiceConnection.off("message", messageHandler);
+		};
+	}, [backgroundServiceConnection, foundDevtools, setFoundDevtools]);
```

Comment:
When we chatted we concluded that having `foundDevtools` here would make this run multiple times unnecessarily, no? Since this effect is the only place that can change it in the first place, feels like it shouldn't need to have a "dependency" on it (like it would if some other piece of code could change the value, then this would need to re-run). And we can start the timer without any conditions.